### DISCUSSION TITLE
Continue calling onDrag events even after the pointerout event

### DISCUSSION THREAD

**Discussion by brian-byassee** [katspaugh/wavesurfer.js#3476] [Ideas]
I updated the draggable.ts file here to support drag events even after the cursor has left the wave surfer. I think this makes the seekToDrag function more intuitive and inline with what users might expect when seeking through a video or audio file.

The thing is this could be a breaking change for applications that are expecting the pointerout event to fire onPointerUp.

Thoughts on if this changes feels safe enough or if this just isnt the sort of functionality wanted in wavesurfer.js

Here is the PR preview:
[https://github.com/katspaugh/wavesurfer.js/compare/main...brian-byassee:wavesurfer.js:waversurfer-draggable-update?expand=1](https://github.com/katspaugh/wavesurfer.js/compare/main...brian-byassee:wavesurfer.js:waversurfer-draggable-update?expand=1)

**Comment by brian-byassee** [comment]
Closing as I have found issues with this current update with envelope

**Comment by katspaugh** [comment]
That event is assigned to the document object, so unless your waveworm is close to the edge of the page, it's supposed to continue dragging even outside of the waveform. Perhaps the listener isn't added correctly though. We should check that it fires only on the document object and not on a child element. It might need `{ capture: true }`.

**Reply by brian-byassee** [reply]
Steps to reproduce the issue would be.
1. https://wavesurfer.xyz/examples/?basic.js
2. Add `dragToSeek: true`
3. Start dragging and then with the mouse still down drag below the waveform. The drag event gets canceled.

> That event is assigned to the document object

I think this is because 'document' is the wavesurfer shadow root, not document.body of the page.

Additionally `pointerout` calls `onPointerUp`

**Reply by katspaugh** [reply]
I've just fixed this in #3477.

**Reply by brian-byassee** [reply]
Ah brilliant. Much better than the approach I had started down. Thank you 🙏 

And sorry I thought I might save you time and yet it seems I probably took more of your time.

**Reply by katspaugh** [reply]
Thanks! I appreciate your pointing out this problem, I think it’s good we fixed it.
